src/Sulu/Component/Content/Types/Rlp/Strategy/RlpStrategy.php

```diff
@@ -184,61 +221,42 @@ private function adaptResourceLocators(ResourceSegmentBehavior $document, $userI
         $webspaceKey = $this->documentInspector->getWebspace($document);
         $languageCode = $this->documentInspector->getLocale($document);
 
-        foreach ($document->getChildren() as $childDocument) {
-            $childNode = $this->documentInspector->getNode($childDocument);
-
-            // determine structure
-            $templatePropertyName = $this->nodeHelper->getTranslatedPropertyName('template', $languageCode);
+        $node = $this->documentInspector->getNode($document);
+        $node->getSession()->save();
 
-            if (!$childNode->hasProperty($templatePropertyName)) {
+        foreach ($document->getChildren() as $childDocument) {
+            // skip documents without assigned resource segment
+            if (!$childDocument instanceof ResourceSegmentBehavior
```

Comment:
Is this check needed at all? It wasn't there before, and it's the reason this if is so complicated.